src/date/expand-pattern.js

```diff
@@ -33,10 +55,43 @@ return function( pattern, cldr ) {
 
 	switch ( true ) {
 		case "skeleton" in pattern:
-			result = cldr.main([
-				"dates/calendars/gregorian/dateTimeFormats/availableFormats",
-				pattern.skeleton
-			]);
+			var flagDate = false, flagTime = false;
+			var skeleton = pattern.skeleton;
+			for ( var i = 0 ; i < skeleton.length ; i++ ) {
+				var ch = skeleton[ i ];
+				if ( ch == 'h' || ch == 'H' || ch == 'm' || ch == 's' ) { //Check if skeleton contains Time component
+					flagTime = true;
+				}
+				if ( ch == 'y' || ch == 'M' || ch == 'd' ) { // Check if skeleton contains Date component
+					flagDate = true;
+				}
```

Comment:
> If a client-requested set of fields includes both date and time fields

1. What are all the available date fields?
1. What are all the available time fields?
1. A could be written as B:

A:
```js
ch == 'h' || ch == 'H' || ch == 'm' || ch == 's'
```

B:
```js
/[hHms]/.test( ch )
```